### ISSUE TITLE
make message stash optional

### ISSUE THREAD

**Issue by oswaldo** [darienmt/kamon-logstash#3]
Hello,

Doing some tests I've found that case the logstash server is unreachable for some reason, the current implementation would simply stash everything within the MetricShipper actor, which in the worst case leads to an OutOfMemoryException.
As it is not expected that a monitoring solution impact the monitored system, I would expect stashing messages to be optional at least.
Architecturally there are different ways to model this option, but as I don't have the real need for it and I'm new to this library, maybe @darienmt would have a better/quicker to implement idea.
Meanwhile in my fork [I just disabled that](https://github.com/darienmt/kamon-logstash/compare/master...oswaldo:master) completely.

Cheers,
Oswaldo

**Comment by darienmt** [comment]
Hi Oswaldo,
I agree with you the monitoring solution should not impact the system performance. This is almost impossible because observing a system most of the time change its state. In order to minimize this impact, I would try to separate the logstash ingestions from the monitoring with a Kafka topic. This solution will scale better as Kafka is capable of accepting more load. In future releases, I had plans to implement this part with another module. Dropping the metric could be more restricting than what you need.

Cheers,

Darien

**Comment by alienscience** [comment]
My feeling is that stashing without limit is a dangerous bug. Its bad if logstash goes down but its really bad if this takes down a customer facing system.

@darienmt I would be really grateful if you would reconsider the behaviour.

**Comment by darienmt** [comment]
I agree it is a less than recommended behavior to stash without limit. At the time when I put my comment, I was not considering it properly. I would prefer to have a limit different than zero to accommodate smaller disruption on the sink availability.
